frontend/main.go

```diff
@@ -145,16 +146,24 @@ func main() {
 		log.Fatalf("Error creating Kubernetes client: %s", err)
 	}
 
-	// Start server
-	r, err := startHTTPServer(&flags)
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
-	}
-
-	log.Println("Starting Odigos UI...")
-	log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
-	err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
+	// Attempt to start the server on port 3000 and increment the port number if it's already in use
+	for {
+		r, err := startHTTPServer(&flags)
+		if err != nil {
+			log.Printf("Error starting server on port %d: %s", flags.Port, err)
+			flags.Port++ // Try the next port
+		} else {
+			log.Printf("Starting Odigos UI...")
+			log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
+			err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
+			if err != nil {
+				log.Printf("Error starting server on port %d: %s", flags.Port, err)
+				flags.Port++ // Try the next port
+			} else {
+				// Server started successfully				
+				break
+			}
```

Comment:
nit: The `r.Run` will actually block until the server is stopped:
```
// Note: this method will block the calling goroutine indefinitely unless an error happens.
```

So the comment is not accurate in this case.